tests/test_diffusion.py

```diff
@@ -123,7 +123,7 @@ def execute_devito(ui, dx=0.01, dy=0.01, a=0.5, timesteps=500):
     dt = dx2 * dy2 / (2 * a * (dx2 + dy2))
     # Allocate the grid and set initial condition
     # Note: This should be made simpler through the use of defaults
-    u = TimeData(name='u', shape=(nx, ny), time_order=1)
+    u = TimeData(name='u', shape=(nx, ny), time_order=1, space_order=2)
```

Comment:
I agree in theory, but for practical purposes this was the fastest way to achieve a clean API. For the order(s) to become part of the operator we would have to make the derivative symbols behave like place holders, and only start expanding them once we actually build the operator. This is definitely possible but would require yet another significant re-write, I guess. Another option would have been to make `dx2` and friends functions and allow the order to be overridden in the call, but I thought this looks cleaner.

Also, for practical purposes the `space_order` and `time_order` attributes on `TimeData` objects can freely be adjusted in-between equation/operator instantiations, so while it may not be the most obvious way, the use case you describe is covered.